### DISCUSSION TITLE
If I want to cut my own rigid body model in Unity3d, how do I do it?

### DISCUSSION THREAD

**Discussion by MingSongJiang** [InfinyTech3D/SofaUnity#114] [Help - Using SofaUnity] [answered]
Hello,
If I want to cut my own rigid body model in Unity3d, how do I do it?
In the demo, the cut at the carving is rough, and the cut is not closed.And no matter where I put the cube, the ball cuts in the same place.
Sofa:V23.12
Unity:2020.3.32

![2](https://github.com/user-attachments/assets/f9bcd856-968a-40f9-b275-fdff61d3706d)
![3](https://github.com/user-attachments/assets/918afbbf-7ee0-4e4b-8364-1a58fdcf6325)

Best regards.

**Comment by epernod** [comment, marked as answer]
Hi,

to carve your own model you need to edit the SOFA .scn file and change the mesh. Check SOFA documentation on the scene creation for more help.

Then this example is demonstrating the CarvingManager. Carving is different from cutting, it directly remove element, not cut them. This means you are removing/deleting material from your 3D mesh.

Finally, you need to move the sphere using the SofaObjectController, not the Unity transform. 
![image](https://github.com/user-attachments/assets/73898163-64b8-4c9e-81b7-aef424c4b03d)

There are short key to move it:
- keys 4, 6 for left right
- keys 2, 8 for backward/frontward
- keys 5 and 0 for up and down
- space to carve

like for the LiverInteraction: https://infinytech3d.com/examples-interactions/
